### ISSUE TITLE
failed to inference on single input

### ISSUE THREAD

**Issue by chychen** [NVIDIA/NeMo#2638]
https://github.com/NVIDIA/NeMo/blob/c9d04851e8a9c1382326862126788fadd77663ac/nemo/collections/common/losses/spanning_loss.py#L64

if we inference on single input data, the squeeze function will convert a list to a scalar.

take question_answering for example, it will fail when inferring on a single input.

**Comment by titu1994** [comment]
https://github.com/NVIDIA/NeMo/blob/c9d04851e8a9c1382326862126788fadd77663ac/nemo/collections/common/losses/spanning_loss.py#L44

Start logits neural type shape is (B, T), so the input shape should be (1, T) for a single sample.

**Comment by chychen** [comment]
@titu1994 thanks for helping out, and sorry that my last comment is wrong. please see the code below where I believe is the root cause `s.squeeze()` resulting question_answering failed for a single sample.

https://github.com/NVIDIA/NeMo/blob/564c67df7ffc16cd34629ecf06f505e7f82df174/nemo/collections/nlp/models/question_answering/qa_model.py#L242

### when for multiple samples (num_samples)

`before s.squeeze()`
torch.Size([num_samples, max_sequence, 1])
`afters s.squeeze()`
torch.Size([num_samples, max_sequence])

### when for a single sample (num_samples=1)

`before s.squeeze()`
torch.Size([1, max_sequence, 1])
`afters s.squeeze()`
torch.Size([max_sequence])

**Comment by titu1994** [comment]
@ekmb any suggestion here? Is this a bug or expected result?

**Comment by ekmb** [comment]
@yzhang123 could you take a look at this?

**Comment by yzhang123** [comment]
https://github.com/NVIDIA/NeMo/pull/2668
